crates/holochain/src/core/share.rs

```diff
@@ -0,0 +1,90 @@
+use std::sync::Arc;
+
+pub type ShareResult<T> = Result<T, ShareError>;
+
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
+pub enum ShareError {
+    /// This object is closed, calls on it are invalid.
+    #[error("This object is closed, calls on it are invalid.")]
+    Closed,
+    /// An error ocurred in a closure that accessing the Share state
+    #[cfg(feature = "unstable-countersigning")]
+    #[error(transparent)]
+    ClosureFailed(Box<dyn std::error::Error + Send + Sync>),
```

Comment:
As written, this is incorrect encapsulation, using this for a countersigning session failure. We could rename this "Other" for a generic error reporting, but since its not used anywhere in *this* type's code, I'd vote instead for picking a different error type to be used down in the countersigning code.